commands/command_pull.go

```diff
@@ -0,0 +1,34 @@
+package commands
+
+import (
+	"github.com/github/git-lfs/git"
+	"github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra"
+)
+
+var (
+	pullCmd = &cobra.Command{
+		Use:   "pull",
+		Short: "Downloads LFS files for the current ref, and checks out",
+		Run:   pullCommand,
+	}
+)
+
+func pullCommand(cmd *cobra.Command, args []string) {
+
+	ref, err := git.CurrentRef()
+	if err != nil {
+		Panic(err, "Could not pull")
+	}
+
+	// Previously we would only checkout files that were downloaded, as they
+	// were downloaded. However this would ignore files where the content was
+	// already present locally (since these are no longer included in transfer Q for
+	// better reporting purposes).
+	// So now we do exactly what we say on the tin, fetch then a separate checkout
```

Comment:
I dunno, I find source comments head off misunderstandings better, since they're right there when you wonder why something is done that way. People don't always want to start digging through old PRs / commits to find explanations when they think they've spotted something, I've seen many an accidental regression that way.